client/my-sites/earn/components/add-edit-coupon-modal/test/products-selector.jsx

```diff
@@ -0,0 +1,210 @@
+/**
+ * @jest-environment jsdom
+ */
+
+const mockUseTranslate = () => ( text ) => text;
+const mockTranslate = ( text ) => text;
+
+jest.mock( 'i18n-calypso', () => ( {
+	...jest.requireActual( 'i18n-calypso' ),
+	useTranslate: jest.fn(),
+	translate: jest.fn(),
+} ) );
+jest.mock( '@wordpress/i18n', () => ( {
+	...jest.requireActual( '@wordpress/i18n' ),
+	__: jest.fn(),
+} ) );
+
+import { act, render, screen } from '@testing-library/react';
+import { __ } from '@wordpress/i18n';
+import { translate, useTranslate } from 'i18n-calypso';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import ProductsSelector from '../products-selector';
+
+const testProduct1 = {
+	ID: 9,
+	currency: 'USD',
+	price: 5,
+	title: 'Monthly Subscription',
+	interval: '1 month',
+	buyer_can_change_amount: false,
+	multiple_per_user: false,
+	welcome_email_content: 'Welcome!',
+	subscribe_as_site_subscriber: true,
+	renewal_schedule: '1 month',
+	type: 'newsletter',
+	is_editable: false,
+};
+
+const testProduct2 = {
+	ID: 10,
+	currency: 'USD',
+	price: 40,
+	title: 'Yearly Subscription',
+	interval: '1 year',
+	buyer_can_change_amount: false,
+	multiple_per_user: false,
+	welcome_email_content: 'Welcome!',
+	subscribe_as_site_subscriber: true,
+	renewal_schedule: '1 year',
+	type: 'newsletter',
+	is_editable: false,
+};
```

Comment:
If some of the properties are the same, can we declare them in a separate object and reuse them?

```suggestion
const productData = {
	currency: 'USD',
	buyer_can_change_amount: false,
	multiple_per_user: false,
	welcome_email_content: 'Welcome!',
	subscribe_as_site_subscriber: true,
	type: 'newsletter',
	is_editable: false,
};

const testProduct1 = {
	...productData,
	ID: 9,
	price: 5,
	title: 'Monthly Subscription',
	interval: '1 month',
	renewal_schedule: '1 month',
};

const testProduct2 = {
	...productData,
	ID: 10,
	price: 40,
	title: 'Yearly Subscription',
	interval: '1 year',
	renewal_schedule: '1 year',
};

```